Trying to do Windows Implementation of Vmone

ctoabidmaqbool opened the pull request:
I am done with most of the things!

Yet, there are some little problem, which need to be fixed!

1. [gvm-24-1-windows-x64.zip](https://github.com/gluonhq/mobile/releases/download/gvm-24-1/windows-x64.zip) contains absolute path which can't be easily extract with `ar.exe` or `lib.exe`. As there is no such drive `D:/` in github actions runner (windows container)
   ```
   $ ar t libjdk.lib
   d:/a/mobile/mobile/build/windows-x64/support/native/java.base/libjava/static/AccessController.obj
   d:/a/mobile/mobile/build/windows-x64/support/native/java.base/libjava/static/Array.obj
   d:/a/mobile/mobile/build/windows-x64/support/native/java.base/libjava/static/BootLoader.obj
   ...
   ```
2. The linux alternative [gvm-24-1-linux-x64.zip](https://github.com/gluonhq/mobile/releases/download/gvm-24-1/linux-x64.zip) contains only `.o` files not `*.lib.res` so shuld these files will be added up to final `vmone.lib` for windows?
3. There are almost _657_ files in **libjdk.a** file but _1016_ files are in **libjdk.lib**, why?
4. Even in my local mashine I am able to get [vmone.lib](https://github.com/ctoabidmaqbool/vmone-fork/blob/windows-impl/tmp/vmone.lib), But still It have many issues:

[process-compile-1726634934329.log](https://github.com/user-attachments/files/17042431/process-compile-1726634934329.log)
[process-link-1726634936148.log](https://github.com/user-attachments/files/17042432/process-link-1726634936148.log)

I alreay have disscussed most of the things here: https://github.com/gluonhq/substrate/pull/1279#issuecomment-2357849448

ctoabidmaqbool committed:
Temporary change from tags to branches to do get helps form github actions runner on evey push!

ctoabidmaqbool committed:
Enable windows github actions and debug uname in windows

ctoabidmaqbool committed:
debug if libjdk.lib is copied to correct directory and we are in correct directory!

ctoabidmaqbool committed:
use correct uname e.g. MSYS_NT-10.0-20348

ctoabidmaqbool committed:
try to verify and make sure in both linux and window same jdk must be used!

ctoabidmaqbool committed:
mistakenly changed version for build-android rather then for build-windows-x64 bit

ctoabidmaqbool committed:
try to fix use jdk 11 in windows too

ctoabidmaqbool committed:
try to setup msy2, e.g. D:\msys64\usr\include\sys\mman.h

ctoabidmaqbool committed:
Custom install MSYS2

ctoabidmaqbool committed:
fix no such!

ctoabidmaqbool committed:
try to fix: D:\a\_temp\b47e13e7-f3b6-4bed-8d8a-3e94a4e725b3.sh: line 1: C:toolsmsys64usrbinbash.exe: command not found

ctoabidmaqbool committed:
fixed, \usr\include\sys not exists!

ctoabidmaqbool committed:
Another try using msys2_shell.cmd

ctoabidmaqbool committed:
as Online is git bash shell, some debugging

ctoabidmaqbool committed:
set system path variable using different method!

ctoabidmaqbool committed:
github path syntax, fix

ctoabidmaqbool committed:
trying out $PATH instead of $GITHUB_PATH

ctoabidmaqbool committed:
path variable using different way

ctoabidmaqbool committed:
using bash

ctoabidmaqbool committed:
fix path ls /c/tools/msys64/usr/include/sys

ctoabidmaqbool committed:
try to use msys64 bash terminal

ctoabidmaqbool committed:
fix path syntax,
and remove all others jobs except windwos-64 bit to focus on windows only!

ctoabidmaqbool committed:
try to install necessory nsys64 PACKAGES

ctoabidmaqbool committed:
switch to MINGW64_NT-10.0-20348

ctoabidmaqbool committed:
switch to msys64 for make clean all

ctoabidmaqbool committed:
Run make clean all in bash shell!

ctoabidmaqbool committed:
bash without --login -i

ctoabidmaqbool committed:
try to match MINGW64_NT without version info!

ctoabidmaqbool committed:
try to add java before make clean build

ctoabidmaqbool committed:
witch shell to msys64 directly!

ctoabidmaqbool committed:
USING msys2/setup-msys2@v2

ctoabidmaqbool committed:
Fix, MINGW64_NT-10.0-20348 name

ctoabidmaqbool committed:
use 64 bit jdk

ctoabidmaqbool committed:
fix jdk path:

ctoabidmaqbool committed:
Another little path fix!

ctoabidmaqbool committed:
Another test if java available!

ctoabidmaqbool committed:
Fix some syntax

ctoabidmaqbool committed:
fix another syntax

ctoabidmaqbool committed:
use long long

ctoabidmaqbool committed:
For windows switch to AR from Lib!

ctoabidmaqbool committed:
fix correct upload path!

ctoabidmaqbool committed:
Optiomized the github workflows file a little bit!

ctoabidmaqbool committed:
Extra file and move using using msys2 shell instead of powershell way!

ctoabidmaqbool committed:
Install unzip utility of msys2 package

ctoabidmaqbool committed:
Again apply others too othere then windows-x64

ctoabidmaqbool committed:
Github Actions again swith to tags instead of branches.
Incluse other files too other then *.obj when creating windows build.
Remove extra files, which was added mistakenly.

ctoabidmaqbool committed:
Merge remote-tracking branch 'remotes/origin/main' into windows-impl

ctoabidmaqbool committed:
Switch to actions/setup-java@v4

ctoabidmaqbool committed:
rename libvmone.lib to vmone.lib as substrate is looking for vmone.lib in case of windows

ctoabidmaqbool committed:
Do some debugging why .objs are not including in the final .lib!
Enable both branches and tages commit to run github actions;

ctoabidmaqbool committed:
trying to do debugging of make

ctoabidmaqbool committed:
Try to do partial match: e.g. MINGW64_NT
Fixed: In case of windows from libjdk.lib only files must be added not folders too!

ctoabidmaqbool committed:
debugging files extracted by libjdk.lib

ctoabidmaqbool committed:
debugging if d drive is available online;

ctoabidmaqbool committed:
some more tests

ctoabidmaqbool committed:
debugging some more!

ctoabidmaqbool committed:
debugging if folders are createing?

ctoabidmaqbool committed:
fix code formate

ctoabidmaqbool committed:
fix some more syntax erros

ctoabidmaqbool committed:
Remove extras:
Try to create D drive;

ctoabidmaqbool committed:
fix syntax

ctoabidmaqbool committed:
check online ar version!

ctoabidmaqbool committed:
try to extract files in temp form libjdk.lib

ctoabidmaqbool committed:
Try to add local finished file!

ctoabidmaqbool commented:
[substrate-fork](https://github.com/ctoabidmaqbool/substrate-fork/tree/windows-impl) to do testing vmone for windows!

jperedadnr commented:
You will get one step closer if you restore something that was removed, probably by mistake, in Substrate:
https://github.com/gluonhq/substrate/commit/f93489a4e6706552c8aeebb9482e2cbbc03f9a37#diff-6671afa2eae21f712d1059fa9bfc797c2c04140ac9c9e5fbe9913fdfc2137abbL50

Just add to the WindowsTargetConfiguration:
```
private static final List<String> javaWindowsLibs = List.of(
            "advapi32", "iphlpapi", "secur32", "userenv",
            "version", "ws2_32", "winhttp", "ncrypt",
            "crypt32", "mswsock"
    );

    @Override
    List<String> getOtherStaticLibs() {
        return javaWindowsLibs;
    }
```
 With that I get down to 15 unresolved symbols...

ctoabidmaqbool committed:
dubbing dirlist.txt;

ctoabidmaqbool commented:
@jperedadnr Thanks, Using your provided help, I have getup much progress!

```
Process
=======
link

Command Line
============
link D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\gluonfx\x86_64-windows\gvm\HelloGluon\launcher.obj D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\gluonfx\x86_64-windows\gvm\tmp\SVM-1726700354384\com.gluonhq.hello.hellogluonapp.obj D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\gluonfx\x86_64-windows\gvm\HelloGluon\IconGroup.obj vmone.lib advapi32.lib iphlpapi.lib secur32.lib userenv.lib version.lib ws2_32.lib winhttp.lib ncrypt.lib crypt32.lib mswsock.lib /NODEFAULTLIB:libcmt.lib /SUBSYSTEM:WINDOWS /ENTRY:mainCRTStartup comdlg32.lib dwmapi.lib gdi32.lib imm32.lib shell32.lib uiautomationcore.lib urlmon.lib winmm.lib javafx_font.lib javafx_iio.lib prism_common.lib /WHOLEARCHIVE:javafx_font.lib /WHOLEARCHIVE:javafx_iio.lib /WHOLEARCHIVE:prism_common.lib prism_sw.lib /WHOLEARCHIVE:prism_sw.lib /OUT:D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\gluonfx\x86_64-windows\HelloGluon.exe /LIBPATH:C:\Users\MSC-30\.gluon\substrate\javafxStaticSdk\24-ea+7.1\windows-x86_64\sdk\lib /LIBPATH:C:\Users\MSC-30\.gluon\substrate\javaStaticSdk\24-4\windows-x86_64\lib

Output
======
Microsoft (R) Incremental Linker Version 14.39.33523.0
Copyright (C) Microsoft Corporation.  All rights reserved.

vmone.lib(awt_Toolkit.obj) : error LNK2005: DllMain already defined in vmone.lib(net_util_md.obj)
   Creating library D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\gluonfx\x86_64-windows\HelloGluon.lib and object D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\gluonfx\x86_64-windows\HelloGluon.exp
vmone.lib(JvmFuncs.o) : error LNK2019: unresolved external symbol sysconf referenced in function JVM_ActiveProcessorCount
vmone.lib(JvmFuncs.o) : error LNK2019: unresolved external symbol __errno referenced in function JVM_Connect
vmone.lib(JvmFuncs.o) : error LNK2019: unresolved external symbol __getreent referenced in function JVM_FindLibraryEntry
vmone.lib(cpuid.o) : error LNK2001: unresolved external symbol __getreent
vmone.lib(dummy.o) : error LNK2001: unresolved external symbol __getreent
vmone.lib(JvmFuncs.o) : error LNK2019: unresolved external symbol dlsym referenced in function JVM_FindLibraryEntry
vmone.lib(JvmFuncs.o) : error LNK2019: unresolved external symbol ioctl referenced in function JVM_SocketAvailable
vmone.lib(JvmFuncs.o) : error LNK2019: unresolved external symbol gettimeofday referenced in function Java_java_lang_System_currentTimeMillis
vmone.lib(JvmFuncs.o) : error LNK2001: unresolved external symbol __svm_vm_is_static_binary
vmone.lib(cpuid.o) : error LNK2019: unresolved external symbol ___chkstk_ms referenced in function checkCPUFeatures
vmone.lib(awt_Toolkit.obj) : error LNK2019: unresolved external symbol JVM_RaiseSignal referenced in function "public: static __int64 __cdecl AwtToolkit::WndProc(struct HWND__ *,unsigned int,unsigned __int64,__int64)" (?WndProc@AwtToolkit@@SA_JPEAUHWND__@@I_K_J@Z)
vmone.lib(awt_Toolkit.obj) : error LNK2019: unresolved external symbol __imp_StrTrimW referenced in function "private: void __cdecl AwtToolkit::InitTouchKeyboardExeFilePath(void)" (?InitTouchKeyboardExeFilePath@AwtToolkit@@AEAAXXZ)
vmone.lib(ComCtl32Util.obj) : error LNK2019: unresolved external symbol __imp_InitCommonControlsEx referenced in function "public: void __cdecl ComCtl32Util::InitLibraries(void)" (?InitLibraries@ComCtl32Util@@QEAAXXZ)
vmone.lib(ComCtl32Util.obj) : error LNK2019: unresolved external symbol SetWindowSubclass referenced in function "public: __int64 (__cdecl*__cdecl ComCtl32Util::SubclassHWND(struct HWND__ *,__int64 (__cdecl*)(struct HWND__ *,unsigned int,unsigned __int64,__int64)))(struct HWND__ *,unsigned int,unsigned __int64,__int64)" (?SubclassHWND@ComCtl32Util@@QEAAP6A_JPEAUHWND__@@I_K_J@Z0P6A_J0I12@Z@Z)
vmone.lib(ComCtl32Util.obj) : error LNK2019: unresolved external symbol RemoveWindowSubclass referenced in function "public: void __cdecl ComCtl32Util::UnsubclassHWND(struct HWND__ *,__int64 (__cdecl*)(struct HWND__ *,unsigned int,unsigned __int64,__int64),__int64 (__cdecl*)(struct HWND__ *,unsigned int,unsigned __int64,__int64))" (?UnsubclassHWND@ComCtl32Util@@QEAAXPEAUHWND__@@P6A_J0I_K_J@Z3@Z)
vmone.lib(ComCtl32Util.obj) : error LNK2019: unresolved external symbol DefSubclassProc referenced in function "public: __int64 __cdecl ComCtl32Util::DefWindowProcW(__int64 (__cdecl*)(struct HWND__ *,unsigned int,unsigned __int64,__int64),struct HWND__ *,unsigned int,unsigned __int64,__int64)" (?DefWindowProcW@ComCtl32Util@@QEAA_JP6A_JPEAUHWND__@@I_K_J@Z0I12@Z)
vmone.lib(awt_DnDDT.obj) : error LNK2019: unresolved external symbol __imp_SHCreateStreamOnFileW referenced in function "protected: virtual long __cdecl AwtDropTarget::SaveIndexToFile(unsigned short const *,unsigned int)" (?SaveIndexToFile@AwtDropTarget@@MEAAJPEBGI@Z)
D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\gluonfx\x86_64-windows\HelloGluon.exe : fatal error LNK1120: 15 unresolved externals


Result
======
result: 1120
```

ctoabidmaqbool commented:
> vmone.lib(awt_Toolkit.obj) : error LNK2005: DllMain already defined in vmone.lib(net_util_md.obj)

This is the problem with `libjdk.lib` ([windows-x64.zip](https://github.com/gluonhq/mobile/releases/download/gvm-24-1/windows-x64.zip)) which was merged to vmone.lib.

What about this?

ctoabidmaqbool commented:
https://github.com/ctoabidmaqbool/substrate-fork/commit/886f29b31692f028fbf9f1c07a3701bccc09359a

@jperedadnr After necessory changes, Now!

```
Process
=======
link

Command Line
============
link D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\gluonfx\x86_64-windows\gvm\HelloGluon\launcher.obj D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\gluonfx\x86_64-windows\gvm\tmp\SVM-1726700354384\com.gluonhq.hello.hellogluonapp.obj D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\gluonfx\x86_64-windows\gvm\HelloGluon\IconGroup.obj vmone.lib advapi32.lib iphlpapi.lib secur32.lib userenv.lib version.lib ws2_32.lib winhttp.lib ncrypt.lib crypt32.lib mswsock.lib shlwapi.lib comctl32.lib /NODEFAULTLIB:libcmt.lib /SUBSYSTEM:WINDOWS /ENTRY:mainCRTStartup comdlg32.lib dwmapi.lib gdi32.lib imm32.lib shell32.lib uiautomationcore.lib urlmon.lib winmm.lib javafx_font.lib javafx_iio.lib prism_common.lib /WHOLEARCHIVE:javafx_font.lib /WHOLEARCHIVE:javafx_iio.lib /WHOLEARCHIVE:prism_common.lib prism_sw.lib /WHOLEARCHIVE:prism_sw.lib /OUT:D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\gluonfx\x86_64-windows\HelloGluon.exe /LIBPATH:C:\Users\MSC-30\.gluon\substrate\javafxStaticSdk\24-ea+7.1\windows-x86_64\sdk\lib /LIBPATH:C:\Users\MSC-30\.gluon\substrate\javaStaticSdk\24-4\windows-x86_64\lib

Output
======
Microsoft (R) Incremental Linker Version 14.39.33523.0
Copyright (C) Microsoft Corporation.  All rights reserved.

vmone.lib(awt_Toolkit.obj) : error LNK2005: DllMain already defined in vmone.lib(net_util_md.obj)
   Creating library D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\gluonfx\x86_64-windows\HelloGluon.lib and object D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\gluonfx\x86_64-windows\HelloGluon.exp
vmone.lib(JvmFuncs.o) : error LNK2019: unresolved external symbol sysconf referenced in function JVM_ActiveProcessorCount
vmone.lib(JvmFuncs.o) : error LNK2019: unresolved external symbol __errno referenced in function JVM_Connect
vmone.lib(JvmFuncs.o) : error LNK2019: unresolved external symbol __getreent referenced in function JVM_FindLibraryEntry
vmone.lib(cpuid.o) : error LNK2001: unresolved external symbol __getreent
vmone.lib(dummy.o) : error LNK2001: unresolved external symbol __getreent
vmone.lib(JvmFuncs.o) : error LNK2019: unresolved external symbol dlsym referenced in function JVM_FindLibraryEntry
vmone.lib(JvmFuncs.o) : error LNK2019: unresolved external symbol ioctl referenced in function JVM_SocketAvailable
vmone.lib(JvmFuncs.o) : error LNK2019: unresolved external symbol gettimeofday referenced in function Java_java_lang_System_currentTimeMillis
vmone.lib(JvmFuncs.o) : error LNK2001: unresolved external symbol __svm_vm_is_static_binary
vmone.lib(cpuid.o) : error LNK2019: unresolved external symbol ___chkstk_ms referenced in function checkCPUFeatures
vmone.lib(awt_Toolkit.obj) : error LNK2019: unresolved external symbol JVM_RaiseSignal referenced in function "public: static __int64 __cdecl AwtToolkit::WndProc(struct HWND__ *,unsigned int,unsigned __int64,__int64)" (?WndProc@AwtToolkit@@SA_JPEAUHWND__@@I_K_J@Z)
D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\gluonfx\x86_64-windows\HelloGluon.exe : fatal error LNK1120: 9 unresolved externals


Result
======
result: 1120
```

johanvos commented:
ioctl etc are systemcalls in unix-like OS'es, but they don't exist on Windows. So those parts should have their Windows-specific equivalent.

ctoabidmaqbool commented:
> ioctl etc are systemcalls in unix-like OS'es, but they don't exist on Windows. So those parts should have their Windows-specific equivalent.

Thanks for providing suggetions! I am trying to resolve those unresolved sysmbols!

I have temporarly convert `Makefile` project to Visual Studio 2022 -> C++ Static Library project, and trying to resolve window compatible header file and codding, It's more easy to do work with visual studio build system for windows!

![image](https://github.com/user-attachments/assets/19894bb4-9c51-45d1-b17d-bf18e2d29c67)

ctoabidmaqbool commented:
@johanvos @jperedadnr Please! Review my PR!

I have temporary add `Visual Studio 2022 -> C++ Static Library Project / Solutions`. Which I used localy in my PC to get `vmone.lib` only (without **libjdk.lib included**) for testing purpose, until final solution I can get out!

I have put both files e.g. `vmone.lib` and `libjdk.lib` inside .gluoun to specific folder to do testing e.g.
![image](https://github.com/user-attachments/assets/a200ff7b-de5e-495f-9762-b0f08eae0cc7)

Add **linkerArgs** for `hello-gluon-ci` project: 

```
<linkerArgs>
    <arg>libjdk.lib</arg>
    <arg>/LTCG</arg>
    <arg>/FORCE:MULTIPLE</arg>
</linkerArgs>
```

Add `/FORCE:MULTIPLE` to exclude second duplicate defination (**temporary fix**)
```
libjdk.lib(awt_Toolkit.obj) : error LNK2005: DllMain already defined in libjdk.lib(net_util_md.obj) (error before adding)
libjdk.lib(awt_Toolkit.obj) : warning LNK4006: DllMain already defined in libjdk.lib(net_util_md.obj); second definition ignored (warning after adding)
```

Add `/LTGC` as suggestion is given:
```
vmone.lib(JvmFuncs.obj) : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
```

After testing, my desktop App runs secessfully e.g. `mvnw -Pdesktop gluonfx:build gluonfx:nativerun`!

![image](https://github.com/user-attachments/assets/00ebb004-90f8-499a-bcd8-b7d0078f2c7f)

ctoabidmaqbool committed:
Add windows native codding stuff while make sure, Mac/Linux codding not changed at all;
Add JVM_RaiseSignal(int) function;
Add Visual Studio 2022, C++ Static Library Solutions/Project;

ctoabidmaqbool committed:
Add vmone.lib build through Visual Studio 2022 -> Release -> Build Solutions;

ctoabidmaqbool commented:
I thank, before comment I have done some mistake, I am not sure, now my native app is not running

E.g. 

  ```
  mvn -Pdesktop gluonfx:build (working)
  mvn -Pdesktop gluonfx:nativerun (not working) error: Result for run until end: -1073741819
  ```
  
Console logs: 

```
D:\GraalVM-Projects\gluonhq\hello-gluon-ci>mvn -Pdesktop gluonfx:nativerun
[INFO] Scanning for projects...
[INFO]
[INFO] -------------------< com.gluonhq.samples:hellogluon >-------------------
[INFO] Building HelloGluon 1.0.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- gluonfx-maven-plugin:1.0.25-SNAPSHOT:nativerun (default-cli) @ hellogluon ---
Configuration: ProjectConfiguration{graalPath='D:\Programs\graalvm-java23-windows-amd64-gluon-23+25.1-dev', javaStaticSdkVersion='24-4', javafxStaticSdkVersion='24-ea+7.1', javaVersion=23, graalVersion=23, useJNI=true, useJavaFX=true, usePrismSW=true, enableCheckHash=true, targetTriplet=x86_64-microsoft-windows, hostTriplet=x86_64-microsoft-windows, backend='null', bundlesList=[], resourcesList=[], reflectionList=[], jniList=[], initBuildTimeList=[], runtimeArgsList=[], releaseSymbolsList=null, appName='HelloGluon', releaseConfiguration='ReleaseConfiguration{packageType=null, description='The HelloGluon app', vendor='Gluon', version='null', macAppStore=false, macSigningUserName=null, macAppCategory=null, bundleName='null', bundleVersion='null', bundleShortVersion='null', providedSigningIdentity='null', providedProvisioningProfile='null', skipSigning=false, simulatorDevice='null', appLabel='null', versionCode='null', versionName='null', providedKeyStorePath='null', providedKeyStorePassword='null', providedKeyAlias='null', providedKeyAliasPassword='null'}', mainClassName='com.gluonhq.hello.HelloGluonApp', classpath='D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\classes;D:\.my-m2\repository\com\gluonhq\charm-glisten\6.2.3\charm-glisten-6.2.3.jar;C:\Users\MSC-30\.m2\repository\com\gluonhq\attach\display\4.0.21\display-4.0.21-desktop.jar;D:\.my-m2\repository\com\gluonhq\attach\display\4.0.21\display-4.0.21.jar;C:\Users\MSC-30\.m2\repository\com\gluonhq\attach\lifecycle\4.0.21\lifecycle-4.0.21-desktop.jar;D:\.my-m2\repository\com\gluonhq\attach\lifecycle\4.0.21\lifecycle-4.0.21.jar;C:\Users\MSC-30\.m2\repository\com\gluonhq\attach\statusbar\4.0.21\statusbar-4.0.21.jar;D:\.my-m2\repository\com\gluonhq\attach\statusbar\4.0.21\statusbar-4.0.21.jar;C:\Users\MSC-30\.m2\repository\com\gluonhq\attach\storage\4.0.21\storage-4.0.21-desktop.jar;D:\.my-m2\repository\com\gluonhq\attach\storage\4.0.21\storage-4.0.21.jar;D:\.my-m2\repository\com\gluonhq\attach\util\4.0.21\util-4.0.21.jar;D:\.my-m2\repository\org\openjfx\javafx-base\24-ea+5\javafx-base-24-ea+5-win.jar;D:\.my-m2\repository\org\openjfx\javafx-base\24-ea+5\javafx-base-24-ea+5.jar;D:\.my-m2\repository\org\openjfx\javafx-controls\24-ea+5\javafx-controls-24-ea+5-win.jar;D:\.my-m2\repository\org\openjfx\javafx-controls\24-ea+5\javafx-controls-24-ea+5.jar;D:\.my-m2\repository\org\openjfx\javafx-graphics\24-ea+5\javafx-graphics-24-ea+5-win.jar;D:\.my-m2\repository\org\openjfx\javafx-graphics\24-ea+5\javafx-graphics-24-ea+5.jar'}
[Sat Sep 21 21:17:42 PKT 2024][INFO] ==================== RUN TASK ====================
[Sat Sep 21 21:17:42 PKT 2024][FINE] PB Command for run until end: D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\gluonfx\x86_64-windows\HelloGluon.exe
[Sat Sep 21 21:17:42 PKT 2024][FINE] Start process run until end...
[Sat Sep 21 21:17:43 PKT 2024][FINE] Result for run until end: -1073741819
[Sat Sep 21 21:17:43 PKT 2024][SEVERE] Process run until end failed with result: -1073741819
Check the log files under D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\gluonfx\x86_64-windows\gvm\log
And please check https://docs.gluonhq.com/ for more information.
[Sat Sep 21 21:17:43 PKT 2024][INFO] Logging process [run until end] to file: D:\GraalVM-Projects\gluonhq\hello-gluon-ci\target\gluonfx\log\process-run until end-1726935463885.log
             _______  ___      __   __  _______  __    _
            |       ||   |    |  | |  ||       ||  |  | |
            |    ___||   |    |  | |  ||   _   ||   |_| |
            |   | __ |   |    |  |_|  ||  | |  ||       |
            |   ||  ||   |___ |       ||  |_|  ||  _    |
            |   |_| ||       ||       ||       || | |   |
            |_______||_______||_______||_______||_|  |__|

    Access to the latest docs, tips and tricks and more info on
    how to get support? Register your usage of Gluon Substrate now at

    https://gluonhq.com/activate



[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  10.090 s
[INFO] Finished at: 2024-09-21T21:17:46+05:00
[INFO] ------------------------------------------------------------------------
```

* `hello-gluon-ci\target\gluonfx\log`
  [process-run until end-1726936465780.log](https://github.com/user-attachments/files/17082856/process-run.until.end-1726936465780.log)
  [process-link-1726936458519.log](https://github.com/user-attachments/files/17082857/process-link-1726936458519.log)
  [process-compile-1726936454468.log](https://github.com/user-attachments/files/17082858/process-compile-1726936454468.log)
* `hello-gluon-ci\target\gluonfx\x86_64-windows\gvm\log`
  [client-debug0.1.log](https://github.com/user-attachments/files/17082864/client-debug0.1.log)
  [client-debug0.log](https://github.com/user-attachments/files/17082865/client-debug0.log)

ctoabidmaqbool commented:
It looks like I am just wasting my important time! Unfortunatly no proper response from Gluon team!
It looks like gluon techonolgoy is just a expiremental tool, which somehow works and sometime not!

If someone is using old plugins no support at all, [javafxmobile-plugin](https://github.com/javafxports/javafxmobile-plugin).
Now the case of Window in latest [gluonfx-maven-plugin](https://github.com/gluonhq/gluonfx-maven-plugin)
 no support at all!

What could be expacted in feature!

ctoabidmaqbool committed:
Try to do solve errors caused by:
mingw32-make clean all

ctoabidmaqbool committed:
temp fixes!
